File: io-hotmoka-network/src/main/java/io/hotmoka/network/models/responses/ConstructorCallTransactionFailedResponseModel.java
package io.hotmoka.network.models.responses;

import java.math.BigInteger;

import io.hotmoka.beans.annotations.Immutable;
import io.hotmoka.beans.responses.ConstructorCallTransactionFailedResponse;
import io.hotmoka.network.models.updates.UpdateModel;

@Immutable
public class ConstructorCallTransactionFailedResponseModel extends ConstructorCallTransactionResponseModel {
    /**
     * The amount of gas consumed by the transaction as penalty for the failure.
     */
    public final BigInteger gasConsumedForPenalty;

    /**
     * The fully-qualified class name of the cause exception.
     */
    public final String classNameOfCause;

    /**
     * The message of the cause exception.
     */
    public final String messageOfCause;

    /**
     * The program point where the cause exception occurred.
     */
    public final String where;


    public ConstructorCallTransactionFailedResponseModel(ConstructorCallTransactionFailedResponse response) {
        super(response);

        this.gasConsumedForPenalty = response.gasConsumedForPenalty();
        this.classNameOfCause = response.classNameOfCause;
        this.messageOfCause = response.messageOfCause;
        this.where = response.where;
    }

    public ConstructorCallTransactionFailedResponse toBean() {
        return new ConstructorCallTransactionFailedResponse(
        	classNameOfCause,
            messageOfCause,
            where,
            updates.stream().map(UpdateModel::toBean),
            gasConsumedForCPU,
            gasConsumedForRAM,
            gasConsumedForStorage,
            gasConsumedForPenalty
        );
    }
}

File: io-hotmoka-network-thin-client/io-hotmoka-network-thin-client-js/src/models/responses/ConstructorCallTransactionFailedResponseModel.ts
import {ConstructorCallTransactionResponseModel} from "./ConstructorCallTransactionResponseModel";
import {UpdateModel} from "../updates/UpdateModel";

export class ConstructorCallTransactionFailedResponseModel extends ConstructorCallTransactionResponseModel {

    /**
     * The amount of gas consumed by the transaction as penalty for the failure.
     */
    gasConsumedForPenalty: string

    /**
     * The fully-qualified class name of the cause exception.
     */
    classNameOfCause: string

    /**
     * The message of the cause exception.
     */
    messageOfCause: string

    /**
     * The program point where the cause exception occurred.
     */
    where: string

    constructor(
        updates: Array<UpdateModel>,
        gasConsumedForCPU: string,
        gasConsumedForRAM: string,
        gasConsumedForStorage: string,
        gasConsumedForPenalty: string,
        classNameOfCause: string,
        messageOfCause: string,
        where: string
    ) {
        super(updates, gasConsumedForCPU, gasConsumedForRAM, gasConsumedForStorage)
        this.gasConsumedForPenalty = gasConsumedForPenalty
        this.classNameOfCause = classNameOfCause
        this.messageOfCause = messageOfCause
        this.where = where
    }
}